classes/core/APIRouter.inc.php

```diff
@@ -77,7 +80,16 @@ function route($request) {
 		// Ensure slim library is available
 		require_once('lib/pkp/lib/vendor/autoload.php');
 
-		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $this->getEntity());
+		$sourceFile = null;
+		$entity = $this->getEntity();
+		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $entity[0]);
+
+		if (!empty($entity[1])) {
```

Comment:
Thanks, I see the ambiguity more clearly now -- but I'm still not fond of having a framework concern with a special case for an extra level of depth. What about prefixing back-end handlers with a `_`? I think that was discussed elsewhere.